src/main/java/me/zeroeightsix/kami/module/modules/chat/ChatSanitizer.java

```diff
@@ -0,0 +1,134 @@
+
+
+    @EventHandler
+    public Listener<ClientChatReceivedEvent> listener = new Listener<>(event -> {
+        if (mc.player == null){
+            return;
+        }
+
+        String message = removeUsername(event.getMessage().getUnformattedText());
+        if(isOwn(message) && !sanitizeOwnMessages.getValue()){
+            return;
+        }
+
+        if(suffixes.getValue()){
+            message = sanitizeSuffixes(message);
+        }
+
+        if(fancyChat.getValue()){
+            message = sanitizeFancyChat(message);
+
+            if(message.trim().length() == 0){ // prevent empty messages and associated confusion
+                message = "[Fancychat]";
+            }
+        }
+
+        if(slurs.getValue()){
+            message = sanitize(message, slurMatchers, replacement.getValue().getReplacement());
+        }
+
+        if(swears.getValue()){
+            message = sanitize(message, swearMatchers, replacement.getValue().getReplacement());
+        }
+        String oldMessage = removeUsername(event.getMessage().getUnformattedText());
+        if(!oldMessage.equals(message)){
+
+            if(message.length() > 256){ // I'm not sure if message length matters at this point, but this is just for peace of mind
+                message = message.substring(0, 256);
+            }
+            event.setMessage(new TextComponentString(getUsername(event.getMessage().getUnformattedText()) + " " + message));
+        }
+
+    });
+
+    private String sanitize(String toClean, String[] matchers, String replacement){
+
+        if (!aggressiveFilter.getValue()) {
+            for(String matcher : matchers){
+                toClean = toClean.replaceAll("\\b" + matcher + "|" + matcher + "\\b", replacement); // only check for start or end of a word
+            }
+        } else { // We might encounter the scunthorpe problem, so aggressive mode is off by default.
+            for(String matcher : matchers){
+                toClean = toClean.replaceAll(matcher, replacement);
+            }
+        }
+        return toClean;
+    }
+
+    private static String sanitizeSuffixes(String toClean){
+        return toClean.replaceAll("[\\u23D0|«].+$", "");
+    }
+
+    private static String sanitizeFancyChat(String toClean){
+        return toClean.replaceAll("[^\\u0000-\\u007F]", "");
+    }
+
+    private enum ReplacementOptions {
+        Asterisk("****"), Delete(""), Redact("[Redacted]");
+
+        String replacement;
+
+        ReplacementOptions(String s) {
+            replacement = s;
+        }
+        public String getReplacement() {
+            return replacement;
+        }
+
+    }
+
+    //fairly simple - anarchy servers shouldn't have a chat filter so there isn't an attempt to check for bypasses
+    private static final String[] slurMatchers = new String[]{"nigg(a|er)", "chink", "trann(y|ie)", "kike", "fag(got)?", "retard(ed)?"};
```

Comment:
I'd rather obf these regexes so they're not in the source. I'll do it myself later.

Reply:
why obfuscate the regexes? It's just meant to be a quick and dirty catch for most cases. I highly doubt someone is going to try and specifically bypass this. If thats what you want with this module then I believe there are a few java libraries that do a better job at slur detection than a handful of simple regexes. 😛

Reply:
It doesn't look good to have it in the source as a searchable thing, even if the intention is to remove it from the chat.

Reply:
oohhh I misinterpreted your message. I see your point now. Good idea!